pallene/coder.lua

```diff
@@ -647,13 +647,17 @@ function Coder:init_upvalues()
     end
 
     -- String Literals
+    local literals = {}
     for _, func in ipairs(self.module.functions) do
         for cmd in ir.iter(func.body) do
             for _, v in ipairs(ir.get_srcs(cmd)) do
                 if v._tag == "ir.Value.String" then
                     local str = v.value
-                    table.insert(self.upvalues, coder.Upvalue.String(str))
-                    self.upvalue_of_string[str] = #self.upvalues
+                    if not literals[str] then
```

Comment:
Do we need this `literals` table? Would it be possible to just use `self.upvalue_of_string` instead?